metaflow/plugins/env_escape/client_modules.py

```diff
@@ -135,84 +139,90 @@ def __init__(
-                self._python_executable,
-                self._pythonpath,
-                max_pickle_version,
-                self._config_dir,
-            )
-            atexit.register(_clean_client, self._client)
-
-            # Get information about overrides and what the server knows about
-            exports = self._client.get_exports()
-
-            prefixes = set()
-            export_classes = exports.get("classes", [])
-            export_functions = exports.get("functions", [])
-            export_values = exports.get("values", [])
-            export_exceptions = exports.get("exceptions", [])
-            self._aliases = exports.get("aliases", {})
-            for name in itertools.chain(
-                export_classes,
-                export_functions,
-                export_values,
-                (e[0] for e in export_exceptions),
-            ):
-                splits = name.rsplit(".", 1)
-                prefixes.add(splits[0])
-            # We will make sure that we create modules even for "empty" prefixes
-            # because packages are always loaded hierarchically so if we have
-            # something in `a.b.c` but nothing directly in `a`, we still need to
-            # create a module named `a`. There is probably a better way of doing this
-            all_prefixes = list(prefixes)
-            for prefix in all_prefixes:
-                parts = prefix.split(".")
-                cur = parts[0]
-                for i in range(1, len(parts)):
-                    prefixes.add(cur)
-                    cur = ".".join([cur, parts[i]])
-
-            # We now know all the modules that we can handle. We update
-            # handled_module and return the module if we have it or raise ImportError
-            self._handled_modules = {}
-            for prefix in prefixes:
-                self._handled_modules[prefix] = _WrappedModule(
-                    self, prefix, exports, self._client
-                )
+    def create_module(self, spec):
+        # Return None to use default module creation
+        return None
+
+    def exec_module(self, module):
+        fullname = module.__name__
+
+        self._initialize_client()
+
         canonical_fullname = get_canonical_name(fullname, self._aliases)
-        # Modules are created canonically but we need to return something for any
-        # of the aliases.
-        module = self._handled_modules.get(canonical_fullname)
-        if module is None:
-            raise ImportError
-        sys.modules[fullname] = module
-        return module
+        # Modules are created canonically but we need to handle any of the aliases.
+        wrapped_module = self._handled_modules.get(canonical_fullname)
+        if wrapped_module is None:
+            raise ImportError(f"No module named '{fullname}'")
+
+        # Replace the standard module with the wrapped module in sys.modules
+        sys.modules[fullname] = wrapped_module
```

Comment:
Looking at the code of how this is used in the import system, I believe we shouldn't be adding stuff to `sys.modules`. Maybe we can try to do more in `create_module` and return `wrapped_module` there and then not do anything in `exec_module`. I think `exec_module` is also for reloading which we do not support much right now.